src/pool/srv_pool.c

```diff
@@ -3745,6 +3792,82 @@ ds_pool_evict_handler(crt_rpc_t *rpc)
 	crt_reply_send(rpc);
 }
 
+/**
+ * Send a CaRT message to the pool svc during pool destroy to test and
+ * (if applicable based on force option) evict all open handles on a pool.
+ *
+ * \param[in]	pool_uuid	UUID of the pool
+ * \param[in]	ranks		Pool service replicas
+ * \param[in]	force		If true request all handles be forcibly evicted
+ *
+ * \return	0		Success
+ *		-DER_BUSY	Open pool handles exist and no force requested
+ *
+ */
+int
+ds_pool_svc_check_evict(uuid_t pool_uuid, d_rank_list_t *ranks, uint32_t force)
+{
+	int			 rc;
+	struct rsvc_client	 client;
+	crt_endpoint_t		 ep;
+	struct dss_module_info	*info = dss_get_module_info();
+	crt_rpc_t		*rpc;
+	struct pool_evict_in	*in;
+	struct pool_evict_out	*out;
+
+	D_DEBUG(DB_MGMT, DF_UUID": Destroy pool, inspect/evict handles\n",
+		DP_UUID(pool_uuid));
+
+	rc = rsvc_client_init(&client, ranks);
+	if (rc != 0)
+		D_GOTO(out, rc);
+
+rechoose:
+	ep.ep_grp = NULL; /* primary group */
+	rsvc_client_choose(&client, &ep);
+
+	rc = pool_req_create(info->dmi_ctx, &ep, POOL_EVICT, &rpc);
```

Comment:
Hmm, I thought pool destroy should be driven by pool leader, instead of rank 0. i.e. The rank 0 will send pool destroy RPC to the leader,  then the leader can have a callback during destroy stage, which can be used to stop rebuild etc.

Reply:
Yes, I recall we have talked about that approach in the past. It's unclear to me which approach is better. Currently, rebuild master state should be released from pool_svc_step_down_cb (and pool_svc_drain_cb if needed), whereas rebuild slave state could be released from ds_pool_stop code path.

Reply:
hmm, if the pool leader change will also trigger pool_svc_step_down_cb(), then this will not work for rebuild.